src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DirectoryHelper.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public static class DirectoryHelper
+    {
+        public static IEnumerable<string> GetFilteredFiles(string workspaceDirectory, string searchPattern, IReadOnlyCollection<string> ignoredNames, ILogger? logger)
```

Comment:
I think a comment on this method indicating why it's needed and what it does may be helpful for future readers